src/v/datalake/translation/scheduling_policies.cc

```diff
@@ -384,17 +384,29 @@ ss::future<> fair_scheduling_policy::on_resource_exhaustion(
                  > b.status().memory_bytes_reserved;
       });
 
-    auto num_running = executor.running.size();
     // pick the earliest scheduled translator and force a flush.
+    auto& executable = *executor.running.begin();
     vlog(
       datalake_log.debug,
       "[{}] stopping translator due to memory exhaustion",
-      *executor.running.begin());
-    executor.stop_translation(
-      *executor.running.begin(), translator::stop_reason::oom);
+      executable);
+    executor.stop_translation(executable, translator::stop_reason::oom);
+
+    executable.translator_ptr()->set_finish_translation();
+    const auto translator_id = executable.translator_ptr()->id();
```

Comment:
any particular reason not to use finish_translator() method here, curious.

Reply:
oh yeh that's probably a better idea

Reply:
Just making sure I understand, doesn't finishing the translator mean that parquet sizes are now bounded by the amount of available memory? Whereas before, they'd be bounded by disk scratch space?

Reply:
i think `finish_translator` should be renamed to avoid the term _finish_. in `partition_translator::stop_translation` we throw difference exceptions based on the reason provided to `finish_translator` which will induce different behavior.

however, to your point: it looks to me like `oom` handling also finishes the translator (and it did before, too IIUC), instead of merely flushing. i think there was a discussion with @bharathv on slack about this aspect: in an oom scenario, i don't think we need to finish/upload, we just need to flush to disk to release resources?

Reply:
> however, to your point: it looks to me like oom handling also finishes the translator (and it did before, too IIUC), instead of merely flushing. i think there was a discussion with @bharathv on slack about this aspect: in an oom scenario, i don't think we need to finish/upload, we just need to flush to disk to release resources?

yes exactly.. I'll fix this behavior soon. Currently what we have is a conservative behavior to reset the state.

Reply:
Ah my mistake, thanks for clarifying. SGTM